### PR TITLE
A cli arg to define resource group to collect from

### PR DESCRIPTION
**Pull request by olasaadi99**
* A cli arg to define resource group to collect from
* If user specifies no resource group - collect all resource groups
* Update README to include resource group arg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,9 +57,13 @@ Usage of ./bin/collect:
         cloud provider from which to collect resources
   -region value
         cloud region from which to collect resources
+  -resource-group string
+        resource group id from which to collect resources
+
 ```
 * Value of `-provider` must be either `ibm` or `aws`
 * The `-region` argument can appear multiple times. If running with no `-region` arguments, resources from all regions are collected.
+* If running with no `-resource-group` arguments, resources from all resource groups are collected.
```
Comment (zivnevo) [inline README.md:66]:
```suggestion
* If running with no `-resource-group` argument, resources from all resource groups are collected.

```

File: pkg/ibm/resources_container.go [hunk 2]
```diff
@@ -177,21 +179,21 @@ func (resources *ResourcesContainer) CollectResourcesFromAPI() error {
...
+	err = resources.collectGlobalResources(apiKey, resources.ResourceGroupID)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (resources *ResourcesContainer) collectRegionalResources(region, apiKey string) error {
+func (resources *ResourcesContainer) collectRegionalResources(region, apiKey, resourceGroupID string) error {
```
Comment (zivnevo) [inline pkg/ibm/resources_container.go:196]:
```suggestion
func (resources *ResourcesContainer) collectRegionalResources(region, apiKey string) error {
```
No need to pass `resourceGroupId` as a parameter. Use `resources.ResourceGroupID` instead.

File: pkg/ibm/resources_container.go [hunk 3]
```diff
@@ -273,15 +275,15 @@ func (resources *ResourcesContainer) collectRegionalResources(region, apiKey str
...
+	lbs, err := getLoadBalancers(vpcService, resourceGroupID)
 	if err != nil {
 		return err
 	}
 	resources.LBList = append(resources.LBList, lbs...)
 	return nil
 }
 
-func (resources *ResourcesContainer) collectGlobalResources(apiKey string) error {
+func (resources *ResourcesContainer) collectGlobalResources(apiKey, resourceGroupID string) error {
```
Comment (zivnevo) [inline pkg/ibm/resources_container.go:286]: Same here

File: pkg/ibm/resources_container.go
Comment (zivnevo) [inline pkg/ibm/resources_container.go:1]: Seems like IKS clusters can also be filtered by resource group. `VpcGetClustersOptions` has a field called `XAuthResourceGroup`, which seems to have the same meaning as `ResrouceGroupId` in the VPC API. @kyorav , do you know more?

File: pkg/ibm/vpc_collection.go [hunk 4]
```diff
@@ -45,7 +45,7 @@ func iteratePagedAPI[T any, Q HasGetNextStart[T]](
 }
 
 //nolint:dupl // The duplication is essentially creating the adapter
-func getVPCs(vpcService *vpcv1.VpcV1, region string) ([]*datamodel.VPC, error) {
+func getVPCs(vpcService *vpcv1.VpcV1, region, resourceGroupID string) ([]*datamodel.VPC, error) {
 	APIFunc := func(pageSize int64, next *string) (*vpcv1.VPCCollection, any, error) {
 		return vpcService.ListVpcs(&vpcv1.ListVpcsOptions{Limit: &pageSize, Start: next})
```
Comment (zivnevo) [inline pkg/ibm/vpc_collection.go:50]:
```suggestion
		return vpcService.ListVpcs(&vpcv1.ListVpcsOptions{Limit: &pageSize, Start: next, ResourceGroupId: resourceGroupId})
```
Doesn't it work this way?

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/ibm/vpc_collection.go (hunk 4 above)
Comment (olasaadi99) [reply pkg/ibm/vpc_collection.go:50]: yes it works, I will do the for all of the relevant resources

Comment (olasaadi99) [reply pkg/ibm/vpc_collection.go:50]: but still need to check if the arg is empty

File: pkg/ibm/vpc_collection.go [hunk 5]
```diff
@@ -352,8 +353,9 @@ func getLoadBalancers(vpcService *vpcv1.VpcV1) ([]*datamodel.LoadBalancer, error
 				return nil, err
 			}
 		}
-
-		res[i] = datamodel.NewLoadBalancer(&loadBalancers[i], listeners, pools)
+		if resourceGroupID == "" || *(loadBalancers[i].ResourceGroup.ID) == resourceGroupID {
```
Comment (zivnevo) [inline pkg/ibm/vpc_collection.go:356]: I suggest moving this check to the top of the block, and `continue` if it is does not have the right resource group. This way, you don't collect listeners and pools for skipped LBs

File: pkg/ibm/vpc_collection.go [hunk 6]
```diff
@@ -387,7 +389,8 @@ func getPolicyRules(vpcService *vpcv1.VpcV1, lbID, listenerID string,
 }
 
 //nolint:dupl // See getVPCs
```
Comment (zivnevo) [inline pkg/ibm/vpc_collection.go:391]: I think this nolint is redundant. Try to remove and see if you get lint errors

File: pkg/ibm/vpc_collection.go [hunk 7]
```diff
@@ -399,15 +402,19 @@ func getTransitConnections(tgwService *tgw.TransitGatewayApisV1) ([]*datamodel.T
...
+		for j := range tgwList {
+			if *(transitCons[i].TransitGateway.ID) == *(tgwList[j].ID) {
+				res = append(res, datamodel.NewTransitConnection(&transitCons[i]))
+			}
+		}
 	}
 	return res, nil
 }
 
 //nolint:dupl // See getVPCs
```
Comment (zivnevo) [inline pkg/ibm/vpc_collection.go:416]: Also this nolint

**Review by zivnevo — CHANGES_REQUESTED** [review]
Almost there. Just a few last tiny things

**Review by zivnevo — APPROVED** [review]

**Pull request merged**